infra/modules/providers/azure/aks-appgw/tests/integration/appgateway.go

```diff
@@ -0,0 +1,78 @@
+//  Copyright © Microsoft Corporation
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package integration
+
+import (
+	"os"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-02-01/network"
+	"github.com/microsoft/cobalt/test-harness/infratests"
+	"github.com/microsoft/cobalt/test-harness/terratest-extensions/modules/azure"
+	"github.com/stretchr/testify/require"
+)
+
+var subscription = os.Getenv("ARM_SUBSCRIPTION_ID")
+
+func checkMinCapactiy(t *testing.T, appGatewayProperties *network.ApplicationGatewayPropertiesFormat) {
+	minCapacity := appGatewayProperties.AutoscaleConfiguration.MinCapacity
+	require.Equal(t, int32(2), *minCapacity)
+}
+
+func checkOWASPRuleset(t *testing.T, appGatewayProperties *network.ApplicationGatewayPropertiesFormat) {
+	firewallRulesetType := appGatewayProperties.WebApplicationFirewallConfiguration.RuleSetType
+	firewallRulesetVersion := appGatewayProperties.WebApplicationFirewallConfiguration.RuleSetVersion
+	require.Equal(t, "OWASP", *firewallRulesetType, "Firewall ruleset type is incorrect")
+	require.Equal(t, "3.1", *firewallRulesetVersion, "Firewall ruleset version is incorrect")
+}
+
+func checkAvailablePorts(t *testing.T, appGatewayProperties *network.ApplicationGatewayPropertiesFormat) {
+	foundPort := false
+	frontendPorts := appGatewayProperties.FrontendPorts
+	for _, frontend := range *frontendPorts {
+		if *frontend.Port == int32(443) {
```

Comment:
This ""443" and "3.1" above - will they continue to be this or configurable? If configurable, do we share the test and prod code in some place w.r.t. constants?

Reply:
The integration test for App Gateway is specifically making sure that it can make a request after deployment via https.  It would be a rare situation to expose an alternate port on the public ingress IP then 80 and 443.